tests/RouterTest.php

```diff
@@ -147,12 +147,37 @@ public function testMethodNotAllowedResponses(
     expect(() ==> $router->routeMethodAndPath(HttpMethod::GET, 'headonly'))->toThrow(
       MethodNotAllowedException::class,
     );
+    try {
+      $router->routeMethodAndPath(HttpMethod::GET, 'headonly');
+      static::fail(
+        Str\format('Failed asserting that %s was thrown.', MethodNotAllowedException::class)
+      );
+    } catch(MethodNotAllowedException $e) {
+      expect($e->getAllowedMethods())->toBeSame(keyset[HttpMethod::HEAD]);
+    }
     expect(() ==> $router->routeMethodAndPath(HttpMethod::HEAD, 'postonly'))->toThrow(
       MethodNotAllowedException::class,
     );
+    try {
+      $router->routeMethodAndPath(HttpMethod::HEAD, 'postonly');
+      static::fail(
+        Str\format('Failed asserting that %s was thrown.', MethodNotAllowedException::class)
+      );
+    } catch(MethodNotAllowedException $e) {
+      expect($e->getAllowedMethods())->toBeSame(keyset[HttpMethod::POST]);
+    }
```

Comment:
impl looks good now; for this:

```
$ex = expect(() == > $router->routeMethodAndPath(HttpMethod::HEAD, 'postlnly'))->toThrow(MethodNotAllowedException::class);
expect($ex->getAllowedMethods())->toEqual(keyset[HttpMethod::POST]));
```

`toEqual` is now an alias for `toBeSame` (*not* PHP equality) , and `toBeSame` will be removed

Ditto for the other calls to static::fail

Reply:
fixed :+1: